test/controllers/test-oauth-providers.js

```diff
@@ -0,0 +1,151 @@
+'use strict';
+
+var assert = require('assert');
+var express = require('express');
+var request = require('supertest');
+var uuid = require('uuid');
+var async = require('async');
+
+var helpers = require('../helpers');
+var stormpath = require('../../index');
+
+describe('oauthProviders', function() {
+  var app;
+  var stormpathClient;
+  var stormpathApplication;
+  var response;
+
+  function createStormpathApplication(done) {
+    helpers.createApplication(stormpathClient, function(err, app) {
+      stormpathApplication = app;
+      done(err, app);
+    });
+  }
+
+  function createStormpathExpressApp(done) {
+    app = helpers.createStormpathExpressApp({
+      application: {
+        href: stormpathApplication.href
+      },
+      website: true
+    });
+
+    done(null, app);
+  }
+
+  function waitForReady(done) {
+    app.on('stormpath.ready', done);
+  }
+
+  function performRequest(done) {
+    request(app)
+      .get('/oauth/providers')
+      .end(function(err, result) {
+        if (err) {
+          return done(err);
+        }
+
+        response = JSON.parse(result.text);
+        done(null, response);
+      });
+  }
+
+  before(function(done) {
+    stormpathClient = helpers.createClient();
+
+    async.series([
+      createStormpathApplication,
+      createStormpathExpressApp,
+      waitForReady
+    ], done);
+  });
+
+  after(function(done) {
+    helpers.destroyApplication(stormpathApplication, done);
+  });
+
+  describe('when no social directories exist', function() {
+    before(performRequest);
+
+    it('should return an object', function() {
+      assert.equal(typeof response === 'object', true);
+    });
+
+    it('should have "facebook.enabled" set to false', function() {
+      assert.equal(response.hasOwnProperty('facebook'), true);
```

Comment:
Btw, which assertion library are we using? Because I can't find these functions here: <https://nodejs.org/api/assert.html> :(